[native] Add documentation for the build properties of LinuxMemoryChecker (#23460)
prestodb/presto: presto-docs/src/main/sphinx/presto_cpp/properties.rst

  Number of drivers to use per task. Defaults to hardware CPUs.

Build-enabled Properties
----------------

> The following properties are available only if certain properties are enabled at build time via CMakeLists.txt.

Reviewer: ```suggestion
The following properties are available only if certain properties are enabled at build time by using CMakeLists.txt.
```
Do not use via. See the [GitLab documentation style guide recommended word list entry for via](https://docs.gitlab.com/ee/development/documentation/styleguide/word_list.html#via) for alternatives: I suggested one but please choose another rephrasing if "by using" doesn't work for you.

LinuxMemoryChecker
^^^^^^^^^^^^^^^^^^

Enabling the LinuxMemoryChecker by setting ``PRESTO_ENABLE_LINUX_MEMORY_CHECKER`` 

> to ON in presto-native-execution/CMakeLists.txt will make the following properties available:

Reviewer: ```suggestion
to ``ON`` in presto-native-execution/CMakeLists.txt will make the following properties available:
```
Format values to emphasize them in text.

``systemMemPushbackEnabled``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``boolean``
* **Default value:** ``false``

> If true, starts memory limit checker to trigger memory pushback when

Reviewer: ```suggestion
If set to ``true``, starts memory limit checker to trigger memory pushback when
```

server is under low memory pressure.

``systemMemLimitBytes``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``integer``
* **Default value:** ``0``

Specifies the system memory limit that triggers the memory pushback if
the server memory usage is beyond this limit. This only applies if

> ``systemMemPushbackEnabled`` is true.

Reviewer: ```suggestion
``systemMemPushbackEnabled`` is set to ``true``.
```

``systemMemShrinkBytes``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

* **Type:** ``integer``
* **Default value:** ``0``

Specifies the amount of memory to shrink when the memory pushback is

> triggered. This only applies if ``systemMemPushbackEnabled`` is true.

Reviewer: ```suggestion
triggered. This only applies if ``systemMemPushbackEnabled`` is set to ``true``.
```
